server/src/main/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncAction.java

```diff
@@ -72,4 +85,38 @@ protected void onShardGroupFailure(int shardIndex, Exception exc) {
     protected SearchPhase getNextPhase(final SearchPhaseResults<SearchPhaseResult> results, final SearchPhaseContext context) {
         return new FetchSearchPhase(results, searchPhaseController, context);
     }
+
+    ShardSearchRequest rewriteShardRequest(ShardSearchRequest request) {
```

Comment:
I am not sure I follow. Are you talking of handling queries sorted by _score ? We can probably propagate the global min competitive score up to the query collector so that wouldn't require any rewrite.